src/_lib/modules/word-cloud/word-cloud.component.ts

```diff
@@ -0,0 +1,433 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, EventEmitter, Inject, Input, OnChanges, Output, Renderer2, SimpleChanges } from "@angular/core";
+import { DOCUMENT } from "@angular/common";
+import { IBoundingBox } from "./models/IBoundingBox";
+import { IPoint } from "./models/IPoint";
+import { IWordCloudConfig } from "./models/IWordCloudConfig";
+import { IWord } from "./models/IWord";
+import { IWordWithFontSize } from "./models/IWordWithFontSize";
+import { IWordWithPosition } from "./models/IWordWithPosition";
+import { interval, Observable, of } from "rxjs";
+import { catchError, filter, map, startWith, take, timeout } from "rxjs/operators";
+
+@Component({
+    selector: 'nw-word-cloud',
+    templateUrl: './word-cloud.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class WordCloudComponent<T extends IWord> implements OnChanges {
+
+    @Input() words: T[];
+    @Input() options: Partial<IWordCloudConfig> = {};
+
+    @Output() wordsPositioned: EventEmitter<IWordWithPosition<T>[]> = new EventEmitter();
+
+    private _truncatedWordsWithFontSize: IWordWithFontSize<T>[];
+    private _positionedWords: IWordWithPosition<T>[];
+    private _canvas: HTMLCanvasElement;
+    private _ctx: CanvasRenderingContext2D;
+    private _centerPoint: IPoint;
+    /**
+     * Controls how many points are generated on the spiral. Should be kept between 0 and 1. Lower values generate more
+     * points on the spiral but it can take a lot longer to position the words. If updating this value to <= 0.3, consider
+     * all the places where this is used and for each one the amount of words in the cloud. It could also be a good idea to
+     * make this value configurable by allowing it to be specified in the `options` input
+     */
+    private _spiralResolution = 0.3;
+
+    public config: IWordCloudConfig;
+
+    constructor(
+        private _elRef: ElementRef<HTMLElement>,
+        private _renderer: Renderer2,
+        private _cdRef: ChangeDetectorRef,
+        @Inject(DOCUMENT) private _document: Document,
+        public elRef: ElementRef<HTMLElement>) {}
+
+    ngOnChanges(changes: SimpleChanges): void {
+        const wordsChange: boolean = changes.words?.currentValue !== changes.words?.previousValue;
+        const optionsChange: boolean = changes.options?.currentValue !== changes.options?.previousValue;
+
```

Comment:
What's our general take on typing local variables?  I'm seeing it in some places, but not in others.

Reply:
Local or otherwise we tend to specify the type if it can't be determined by the compiler or if there is any possibility for confusion from a developer point of view. As you're alluding to, I think I'll remove the types here as the compiler can see that they're bools and the `!==` makes it clear enough for other developers I think.

There may be some tsconfig options that could help us be more explicit about when a type is required